### PR TITLE
EZP-26058: update script for non-breaking spaces in ezxmltext

### PR DESCRIPTION
**Pull request by joaoinacio**
JIRA: https://jira.ez.no/browse/EZP-26058

### CODE REVIEW

**Comment by andrerom** [comment]
+0.5

ping @glye / @yannickroger / @dpobel

**Comment by glye** [comment]
+1

Deja Vu - haven't we reviewed this before? :) (The inline fix in kernel/oe maybe.)

I thought the Elvis operator `?:` was still problematic for backports, but it seems even 4.4 requires PHP 5.3, so it should be fine.

**Comment by joaoinacio** [comment]
> I thought the Elvis operator ?: was still problematic for backports, but it seems even 4.4 requires PHP 5.3, so it should be fine.

tbh I copied-pasted that section from another script, but could update it anyway just in case.

File: update/common/scripts/updatenbxmlcontents.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+$limit = array(
+    "offset" => 0,
+    "limit" => $optIterationLimit,
+);
+
+$script->initialize();
+$db = eZDB::instance();
+
+if ( $optDryRun )
+
```
Comment (glye) [inline update/common/scripts/updatenbxmlcontents.php]: `{` missing ;)

**Comment by joaoinacio** [comment]
updated according to @dpobel 's feedback about other tags:
> don't we need to apply the same process in others tag (like heading for instance) ?

AFAICT, heading is the only other tag for which the problem applies.
In any case the tags (`paragraph`, `header`) are now defined in an array first, the regex is only slightly more complex by using named match; see commit https://github.com/ezsystems/ezpublish-legacy/pull/1251/commits/25bac64394e96f22def07fd6d634fb922e6edc3a

**Comment by andrerom** [comment]
+1

File: update/common/scripts/updatenbxmlcontents.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    do {
+        $xmlData = preg_replace( $pattern, $replace, $xmlData, -1, $countReplaced );
+    } while ($countReplaced > 0);
+
+    if ( $verbose ) {
+        eZCLI::instance()->output( "Updating data for content #$contentId (ver. $version) ..." );
+    }
+    if ( !$optDryRun ) {
+        eZDB::instance()->query( "UPDATE ezcontentobject_attribute SET data_text='$xmlData' WHERE id='$id'" );
```
Comment (wizhippo) [inline update/common/scripts/updatenbxmlcontents.php]: this fails if $xmldta contains a ' character.

**Pull request merged**